Solita.LocalizationEditor.UI/DAL/BlobFileAccessStrategy.cs

```diff
@@ -0,0 +1,152 @@
+        }
+        public BlobFileAccessStrategy()
+        {
+
+        }
+        public BlobFileAccessStrategy(IContentRepository contentRepo)
+        {
+            _contentRepository = contentRepo;
+        }
+
+        public override IList<XmlVersionInfo> GetTranslationFileVersions()
+        {
+            return (from mediaFile in GetLocalizationFiles()
+                    select new XmlVersionInfo(mediaFile.BinaryData.ID.ToString(), mediaFile.Name, mediaFile.Created, mediaFile.CreatedBy)
+                    ).ToList();
+        }
+
+        public override XmlDocument LoadXml()
+        {
+            var file = GetLocalizationFile();
+
+            XmlDocument xmlDoc = new XmlDocument();
+            if (file == null)
+            {
+                return xmlDoc;
+            }
+            xmlDoc = LoadBinaryDataToXml(file.BinaryData);
+
+            return xmlDoc;
+        }
+
+        public override XmlDocument LoadVersion(string version)
+        {
+            var localizationFile = GetLocalizationFiles()
+                .Where(f => f.BinaryData != null &&
+                            f.BinaryData.ID != null &&
+                            f.BinaryData.ID.AbsoluteUri.Contains(version))
+                            .SingleOrDefault();
+
+            if (localizationFile == null)
+            {
+                throw new FileNotFoundException($"Could not find file with url: '{version}'");
+            }
+
+            return LoadBinaryDataToXml(localizationFile.BinaryData);
+        }
+
+        public IEnumerable<MediaData> GetLocalizationFiles()
+        {
+            var localizationFileType = typeof(LocalizationFile).Name;
+
+            var files = ContentRepo.GetChildren<MediaData>(LocalizationStorageReference)
+                        .Where(f => f.Name == LocalizationFileName &&
+                            f.GetType().Name == localizationFileType &&
+                            f.BinaryData != null) 
+                            .OrderByDescending(f => f.Created);
+            return files;
+        }
+
+        public MediaData GetLocalizationFile()
+        {
+            var mediaData = GetLocalizationFiles().FirstOrDefault();
+
+            return mediaData ?? ContentRepo.GetDefault<LocalizationFile>(LocalizationStorageReference);
+        }
+
+        public override void SaveXml(XmlDocument xml)
+        {
+            var blobFactory = ServiceLocator.Current.GetInstance<BlobFactory>();
+            var mediaDataResolver = ServiceLocator.Current.GetInstance<ContentMediaResolver>();
+            var contentTypeRepository = ServiceLocator.Current.GetInstance<IContentTypeRepository>();
+
+            var file = ContentRepo.GetDefault<LocalizationFile>(LocalizationStorageReference);
+            file.Name = LocalizationFileName;
+
+            var blob = blobFactory.CreateBlob(file.BinaryDataContainer, ".xml");
+            var xmlBytes = Encoding.UTF8.GetBytes(xml.OuterXml);
```

Comment:
This line seems to be unused. Somehow I'm thinking there could be a more efficient way of saving the XML anyway, using streams instead of reading and parsing the XML as string.